apps/emqx_rule_engine/src/emqx_rule_funcs.erl

```diff
@@ -690,6 +691,16 @@ find_s(S, P, Dir) ->
         SubStr -> SubStr
     end.
 
+jq(FilterProgram, JSONBin)
+  when is_binary(FilterProgram), is_binary(JSONBin) ->
+    case jq:parse(FilterProgram, JSONBin) of
+        {ok, Result} -> erlang:iolist_to_binary(Result);
+        {error, ErrorReason} -> erlang:throw({jq_exception, ErrorReason})
```

Comment:
I changed it to `erlang:error` but I need to look more on how to best deal with error's in the rule engine. @terry-xiaoyu do you have any suggestion?

Reply:
Just use erlang:error/1 here. So `erlang:error({jq_exception, ErrorReason})` is better.

Reply:
Actually, @zmstone wrote https://github.com/emqx/emqx/pull/7384/commits/a219349d5496fd8c17b325a66c92b1092a2582e7#r833943658 before that `erlang:error/1` should not be used unless we want long stack traces. I'm not familiar with the rule engine enough to tell if we really want to throw anything at all here? Will the rule just get ignored, if for some reason the input is not what the user expected so jq gives an error? @qzhuyan @terry-xiaoyu shall we use error or throw or something else given that we get long stack traces if we use error?

Reply:
I have changed back to throw [5cceec2](https://github.com/emqx/emqx/pull/7384/commits/5cceec253bea9b747ec3c3696bf44ab6ba41a237) as I don't think we want to expose long stack traces to the user as @zmstone suggested. Let me know if you think that is wrong and I can change back.

Reply:
Sorry for late reply, I was asking because I didn't see the catcher. I was thinking just return `{error, ErrorInfo}` should be fine.

Anyway, you decide.